kimchi/src/circuits/polynomials/keccak/circuitgates.rs

```diff
@@ -189,6 +189,64 @@ where
     }
 }
 
+//~ ##### `KeccakBits` - 32-bit decomposition gate
```

Comment:
Instead of using an expensive range-check, why don't we just constrain it like this?

```
col 0    1    2 
    x    x_hi x_lo
```
* $x -  x_{hi} \cdot 2^{b} - x_{lo} = 0$ 
* $x_{lo} \cdot 2^{-b} = 0$
* $x_{hi} \cdot 2^{-b} = 0$

where $b = 32$ or $b=64$, as you wish!